test/controllers/document.specs.js

```diff
@@ -0,0 +1,290 @@
+        access: 'user' })
+      .set({ Authorization: `${regularToken}` })
+      .end((err, res) => {
+        expect(res.body.message).to.be.equal('Updated document successfully');
+        expect(res.statusCode).to.be.equal(200);
+        done();
+      });
+    });
+    it('should return error message ', (done) => {
+      request.put('/api/v1/documents/1')
+      .send({ title: 'update my doc',
+        content: 'Updating document ',
+        access: 'user' })
+      .set({ Authorization: `${adminToken}` })
+      .end((err, res) => {
+        expect(res.statusCode).to.be.equal(412);
+        done();
+      });
+    });
+  });
+
+  describe(' Documents search', () => {
+    it('should return all documents that match the search term  ',
+        (done) => {
+          request.get('/api/v1/search/documents/?q=update my doc')
+               .set({ Authorization: adminToken })
+               .end((err, res) => {
+                 expect(res.status).to.be.equal(201);
+                 done();
+               });
+        });
+
+    it('should return 404 error when no document matches the search term  ',
+        (done) => {
+          request.get('/api/v1/search/documents/?q=history')
+               .set({ Authorization: adminToken })
+               .end((err, res) => {
+                 expect(res.status).to.be.equal(404);
+                 done();
+               });
+        });
+  });
+
+  describe(' Delete  document', () => {
+    it(`should return 404 error when no
+       document matches the id parameter supplied`,
+        (done) => {
+          request.delete('/api/v1/documents/1')
+               .set({ Authorization: adminToken })
+               .end((err, res) => {
+                 expect(res.status).to.be.equal(412);
+                 done();
+               });
+        });
+
+    it('should return error message for unauthorize deletion ',
+        (done) => {
+          request.delete('/api/v1/documents/5')
+               .set({ Authorization: regularToken })
+               .end((err, res) => {
+                 expect(res.status).to.be.equal(412);
+                 done();
+               });
+        });
+
+    it('should return  a message when a document is deleted  ',
+        (done) => {
+          request.delete('/api/v1/documents/1')
+               .set({ Authorization: regularToken })
+               .end((err, res) => {
+                 expect(res.status).to.be.equal(200);
+                 expect(res.body.message).to
+                 .be.equal('Deleted document successfully');
+                 done();
+               });
+        });
+
+    it('should return 404 error when user try to access deleted document  ',
+        (done) => {
+          request.get('/api/v1/documents/1')
+               .set({ Authorization: adminToken })
+               .end((err, res) => {
+                 expect(res.status).to.be.equal(201);
+                 done();
+               });
+        });
+  });
+});
```

Comment:
your describe block and it block should clearly spell out the details of the test